src/services/codefixes/importFixes.ts

```diff
@@ -587,19 +606,34 @@ namespace ts.codefix {
         });
     }
 
-    function getExistingImportDeclarations({ moduleSymbol, exportKind, targetFlags, symbol }: SymbolExportInfo, checker: TypeChecker, importingFile: SourceFile, compilerOptions: CompilerOptions): readonly FixAddToExistingImportInfo[] {
-        // Can't use an es6 import for a type in JS.
-        if (!(targetFlags & SymbolFlags.Value) && isSourceFileJS(importingFile)) return emptyArray;
-        const importKind = getImportKind(importingFile, exportKind, compilerOptions);
-        return mapDefined(importingFile.imports, (moduleSpecifier): FixAddToExistingImportInfo | undefined => {
+    function createExistingImportMap(checker: TypeChecker, importingFile: SourceFile, compilerOptions: CompilerOptions) {

```

Comment:
Something about the name of this feels off - what does `existing` refer to?

Reply:
Import declarations that already exist in the file?

Reply:
`createImportMapForExistingExports` then probably

Reply:
Huh? It’s not exports that `existing`, it’s the imports. See historical usage of `ExistingImports` all over this file 😄

Reply:
The purpose of this module is to add imports to a file. By definition, those imports do not exist yet. Contrast them with imports that _do_ exist in the file, which can be used in various ways in the process of adding new ones. Those are what this map contains. It’s a map of imports that exist in the file. We need to specify that they exist since so much in the module deals with hypothetical imports that we might choose to write into existence. Everywhere an existing import is mentioned, it’s called an “existing import,” e.g. the type `FixAddToExistingImport` (a type of fix that adds a new specifier to an existing import declaration). So, I think it’s fair and consistent to say this is a map of existing imports. (More specifically, the keys are the symbol ids of other modules, and the values are the local import declarations that resolve to those modules.)